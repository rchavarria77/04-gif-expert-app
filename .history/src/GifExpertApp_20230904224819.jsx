import React from "react";

const GifExpertApp = () => {
  return (
    <>
      <h1>Gif Expert App</h1>
    </>
  );
};

export default GifExpertApp;
