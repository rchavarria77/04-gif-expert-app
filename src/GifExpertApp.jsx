import { useState } from 'react';
import { AddCategory, GifGrid } from './Components';

const GifExpertApp = () => {
  const [categories, setCategories] = useState(['One Punch']);

  const onAddCategory = (newCategory) => {
    if (categories.find((category) => category.toLowerCase() === newCategory.toLowerCase())) return;

    setCategories([newCategory, ...categories]);
  };

  return (
    <>
      <h1>Gif Expert App</h1>
      <AddCategory onNewCategory={onAddCategory} />

      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};

export default GifExpertApp;
