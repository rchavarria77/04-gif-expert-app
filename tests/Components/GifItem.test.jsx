import { render, screen } from '@testing-library/react';

import { GifItem } from '../../src/Components';

describe('Testing <GifItem /> component', () => {
  const title = 'Dragon Ball';
  const url = 'http://dragon-ball.com/dragon.jpg';

  test('debe de hacer match con el snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });

  test('debe de mostrar la imagen con  el URL y el ALT indicado', () => {
    render(<GifItem title={title} url={url} />);

    // console.log(screen.getByRole('img').src).toBe(url);
    // console.log(screen.getByRole('img').alt).toBe(title);

    const { src, alt } = screen.getByRole('img');

    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('debe de mostrar el titulo en el componente', () => {
    render(<GifItem title={title} url={url} />);

    expect(screen.getByText(title)).toBeTruthy();
  });
});
