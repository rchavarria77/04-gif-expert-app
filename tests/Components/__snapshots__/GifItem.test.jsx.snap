// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifItem /> component debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball"
      src="http://dragon-ball.com/dragon.jpg"
    />
    <p>
      Dragon Ball
    </p>
  </div>
</div>
`;
